Meadowlark-js/backends/meadowlark-mongodb-backend/src/repository/Update.ts

```diff
@@ -3,96 +3,273 @@
+    mongoCollection,
+    session,
+  );
+
+  if (tryUpdateByReplacementResult != null) {
+    // Ensure referenced documents are not modified in other transactions
+    await writeLockReferencedDocuments(mongoCollection, document.outboundRefs, session);
+    return tryUpdateByReplacementResult;
+  }
+
+  // Either the documentUuid doesn't exist or the identity has changed.
+  // The following delete attempt will catch if documentUuid does not exist
+  const deleteResult = await deleteDocumentByIdTransaction(
+    { documentUuid, resourceInfo, security, validateNoReferencesToDocument: true, traceId },
+    mongoCollection,
+    session,
+  );
+  Logger.debug(`${moduleName}.updateAllowingIdentityChange: Updating document uuid ${documentUuid}`, traceId);
+
+  switch (deleteResult.response) {
+    case 'DELETE_SUCCESS':
+      // document was deleted, so we can insert the new version
+      return insertUpdatedDocument(updateRequest, mongoCollection, session, document);
+    case 'DELETE_FAILURE_NOT_EXISTS':
+      // document was not found on delete
+      return { response: 'UPDATE_FAILURE_NOT_EXISTS' };
+    case 'DELETE_FAILURE_REFERENCE':
+      // We have an update cascade scenario
+      //
+      // TODO: Collect all the blocking documents that need updating (not just the first 5!)
+      //       Delete again without validating references
+      //       Do the insert
+      //       If everything succeeds, schedule the cascade - updating each referring document
+      // For now, let it be known this is an update cascade scenario, but do nothing
+      return { response: 'UPDATE_CASCADE_REQUIRED' };
+    default:
+      return { response: 'UNKNOWN_FAILURE', failureMessage: deleteResult.failureMessage };
   }
-  return updateResult;
 }
 
-export async function updateDocumentById(
-  { meadowlarkId, documentUuid, resourceInfo, documentInfo, edfiDoc, validate, traceId, security }: UpdateRequest,
-  client: MongoClient,
+/**
+ * Update the document, disallowing an identity change. Succeeds if a document with the given uuid and
+ * identity (meadowlarkId) exists.
+ *
+ * This function expects Session to have an active transaction. Aborting the transaction on error is left to the caller.
+ */
+async function updateDisallowingIdentityChange(
+  document: MeadowlarkDocument,
+  updateRequest: UpdateRequest,
+  mongoCollection: Collection<MeadowlarkDocument>,
+  session: ClientSession,
 ): Promise<UpdateResult> {
+  // Perform the document update
+  Logger.debug(
+    `${moduleName}.updateDisallowingIdentityChange: Updating document uuid ${updateRequest.documentUuid}`,
+    updateRequest.traceId,
+  );
+
+  // Ensure referenced documents are not modified in other transactions
+  await writeLockReferencedDocuments(mongoCollection, document.outboundRefs, session);
+
+  const tryUpdateByReplacementResult: UpdateResult | null = await tryUpdateByReplacement(
+    document,
+    updateRequest,
+    mongoCollection,
+    session,
+  );
+
```

Comment:
Again, try the common case first. If it doesn't work, then make a second call to find out why.